app/models/event_stream.rb

```diff
@@ -59,11 +59,19 @@ def self.event_groups
 
   def self.group_and_level(event_type)
     level = :detail # the level is detail as default
-    group, _ = event_groups.find do |_k, value|
-      GROUP_LEVELS.detect { |lvl| value[lvl]&.include?(event_type) }.tap do |level_found|
-        level = level_found if level_found
-      end
-    end
+    egroups = event_groups
+
+    group = egroups.find do |_, value|
+      GROUP_LEVELS
+        .find { |lvl| value[lvl]&.select { |typ| typ.kind_of?(String) }&.include?(event_type) }
```

Comment:
`select` is safe, but the `[]` is not. And if I am not mistaken, `&.` operator does not short-circuit, which means that `include?` will still get called on `nil`.

Reply:
And this can be simplified to...

```ruby
GROUP_LEVELS.detect { |lvl| value[lvl]&.any? { |typ| typ.kind_of?(String) && typ == event_type } }
```

Reply:
```ruby
nil&.include?("foo")
# => nil
```

`&.` short-cirtcuits, but my point is that .select can never return nil.  The &. *before* the select must stay as you have it though because value[lvl] can return nil.

Reply:
```
> {a: [1, 2]}[:a].select { |_| false }.include?(1)
=> false
{a: [1, 2]}[:b].select { |_| false }.include?(1)
NoMethodError: private method `select' called for nil:NilClass
{a: [1, 2]}[:b]&.select { |_| false }.include?(1)
NoMethodError: undefined method `include?' for nil:NilClass
```

Now I am almost sure we need that last `&`.